annotation-processor/src/test/java/org/hibernate/validator/ap/testmodel/annotationparameters/InvalidGroupSequenceParameters.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.testmodel.annotationparameters;
+
+import javax.validation.GroupSequence;
+import javax.validation.constraints.AssertFalse;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+/**
+ * @author Marko Bekhta
+ */
+public class InvalidGroupSequenceParameters {
+
+	/**
+	 * Case 1: class present in a list of groups
+	 */
+	public static class Case1 {
+		public interface Group1 {
+		}
+
+		public interface Group2 {
+		}
+
+		@GroupSequence(value = Group2.class)
+		public interface Group3 {
+		}
+
+		public class InvalidGroup1 {
+		}
+
+		public class InvalidGroup2 {
+		}
+
+		@GroupSequence(value = { Group1.class, InvalidGroup1.class })
+		public class SomeBean {
+		}
+
+		@GroupSequence(value = InvalidGroup2.class)
+		public class SomeOtherBean {
+		}
+
+		@GroupSequence(value = { Group3.class, YetOtherBean.class })
+		public class YetOtherBean {
+
+		}
+	}
+
+	/**
+	 * Case 2: primitive present in a list of groups
+	 */
+	public static class Case2 {
+		public interface Group1 {
+		}
+
+		public interface Group2 {
+		}
+
+		@GroupSequence(value = { Group1.class, int.class })
+		public class SomeBean {
+		}
+
+		@GroupSequence(value = boolean.class)
+		public class SomeOtherBean {
+		}
+
+		@GroupSequence(value = double.class)
+		public class YetOtherBean {
+
+		}
+	}
+
+	/**
+	 * Case 3: Cyclic groups
+	 */
+	public static class Case3 {
```

Comment:
I'm a bit confused by this test :) Couldn't a simple cycle test look like this:

    @GroupSequence(value = {Group1.class, Group2.class, Group1.class})
    public interface MySequence {}

I wouldn't consider the case where a group sequence inherits from another interface. It's discouraged by the spec, so if anything, we could raise a warning when detecting it.